napari/layers/_shapes_layer/model.py

```diff
@@ -0,0 +1,1620 @@
+
+        if box is not None:
+            rot = box[1]
+            length_box = np.linalg.norm(box[6] - box[0])
+            if length_box > 0:
+                rescale = self._get_rescale()
+                r = self._rotation_handle_length*rescale
+                rot = rot-r*(box[6] - box[0])/length_box
+            box = np.append(box, [rot], axis=0)
+
+        return box
+
+    def _outline_shapes(self):
+        """Finds outlines of any selected shapes including any shape hovered
+        over
+
+        Returns
+        ----------
+        vertices : None | np.ndarray
+            Nx2 array of any vertices of outline or None
+        triangles : None | np.ndarray
+            Mx3 array of any indices of vertices for triangles of outline or
+            None
+        """
+        if self._hover_shape is not None or len(self.selected_shapes) > 0:
+            if len(self.selected_shapes) > 0:
+                index = copy(self.selected_shapes)
+                if self._hover_shape is not None:
+                    if self._hover_shape in index:
+                        pass
+                    else:
+                        index.append(self._hover_shape)
+                index.sort()
+            else:
+                index = self._hover_shape
+
+            centers, offsets, triangles = self.data.outline(index)
+            rescale = self._get_rescale()
+            vertices = centers + rescale*self._highlight_width*offsets
+        else:
+            vertices = None
+            triangles = None
+
+        return vertices, triangles
+
+    def _compute_vertices_and_box(self):
+        """Compute the location and properties of the vertices and box that
+        need to get rendered
+
+        Returns
+        ----------
+        vertices : np.ndarray
+            Nx2 array of any vertices to be rendered as Markers
+        face_color : str
+            String of the face color of the Markers
+        edge_color : str
+            String of the edge color of the Markers and Line for the box
+        pos : np.ndarray
+            Nx2 array of vertices of the box that will be rendered using a
+            Vispy Line
+        width : float
+            Width of the box edge
+        """
+        if len(self.selected_shapes) > 0:
+            if self.mode == Mode.SELECT:
+                box = self._selected_box[BOX_WITH_HANDLE]
+                if self._hover_shape is None:
+                    face_color = 'white'
+                elif self._hover_vertex is None:
+                    face_color = 'white'
+                else:
+                    face_color = self._highlight_color
+                edge_color = self._highlight_color
+                vertices = box
+                # Use a subset of the vertices of the interaction_box to plot
+                # the line around the edge
+                pos = box[BOX_LINE_HANDLE]
+                width = 1.5
+            elif self.mode in ([Mode.DIRECT, Mode.ADD_PATH, Mode.ADD_POLYGON,
```

Comment:
I think here we would want a comment with each clause, "if the mode is in this class of modes, we only want to render the corner vertices. If the mode is in this other class, we want to render all the vertices of the polygon." Currently it's hard for me to understand what is happening within each block.

Reply:
comments added